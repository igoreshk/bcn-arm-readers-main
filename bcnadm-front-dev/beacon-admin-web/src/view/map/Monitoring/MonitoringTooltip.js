import React from 'react';
import PropTypes from 'prop-types';
import { withLocalize } from 'react-localize-redux';

const rootStyle = {};

const headerStyle = {
  display: 'flex',
  alignItems: 'center'
};

const nameStyle = {
  marginLeft: '8px',
  fontSize: '15px',
  fontWeight: 'bold'
};

const vitalContainerStyle = {
  marginTop: '8px'
};

const vitalHeaderStyle = {
  fontWeight: 'bold'
};

const vitalDetailsStyle = {};

export const MonitoringTooltip = (props) => {
  const { marker, translate } = props;
  const visitorId = marker.visitorId;
  const name = marker.name || '';
  const heartRate = marker.heartRate
    ? `${marker.heartRate} ${translate('monitoringTooltip.bpm')}`
    : translate('monitoringTooltip.unknown');
  const bodyTemperature = marker.bodyTemperature
    ? `${marker.bodyTemperature} Â°C`
    : translate('monitoringTooltip.unknown');
  const stepCount = marker.stepCount
    ? `${marker.stepCount} ${translate('monitoringTooltip.steps')}`
    : translate('monitoringTooltip.unknown');

  return (
    <div style={rootStyle}>
      <div style={headerStyle}>
        <span style={nameStyle}>{name}</span>
        <span style={nameStyle}>{visitorId}</span>
      </div>
      <div style={vitalContainerStyle}>
        <div style={vitalHeaderStyle}>{translate('monitoringTooltip.vital')}</div>
        <div style={vitalDetailsStyle}>
          <div>
            {' '}
            {translate('monitoringTooltip.heart')} {heartRate}
          </div>
          <div>
            {' '}
            {translate('monitoringTooltip.body')} {bodyTemperature}
          </div>
          <div>
            {' '}
            {translate('monitoringTooltip.step')} {stepCount}
          </div>
        </div>
      </div>
    </div>
  );
};

MonitoringTooltip.propTypes = {
  translate: PropTypes.func,
  marker: PropTypes.object.isRequired
};

export default withLocalize(MonitoringTooltip);
