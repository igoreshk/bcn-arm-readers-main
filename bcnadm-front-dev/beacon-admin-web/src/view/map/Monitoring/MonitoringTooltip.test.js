import React from 'react';
import { shallow } from 'enzyme';

import { MonitoringTooltip } from './MonitoringTooltip';

const props = {
  translate: jest.fn(),
  marker: {
    visitorId: 123,
    firstName: 'Mikle',
    lastName: 'Prog'
  }
};

describe('MonitoringTooltip', () => {
  const wrapper = shallow(<MonitoringTooltip {...props} />);

  it('renders', () => expect(wrapper.exists()).toBe(true));

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
