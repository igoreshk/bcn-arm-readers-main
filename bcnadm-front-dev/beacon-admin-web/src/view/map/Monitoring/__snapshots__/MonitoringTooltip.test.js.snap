// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitoringTooltip Check matching snapshot 1`] = `
<div
  style={Object {}}
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <span
      style={
        Object {
          "fontSize": "15px",
          "fontWeight": "bold",
          "marginLeft": "8px",
        }
      }
    />
    <span
      style={
        Object {
          "fontSize": "15px",
          "fontWeight": "bold",
          "marginLeft": "8px",
        }
      }
    >
      123
    </span>
  </div>
  <div
    style={
      Object {
        "marginTop": "8px",
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={Object {}}
    >
      <div>
         
         
      </div>
      <div>
         
         
      </div>
      <div>
         
         
      </div>
    </div>
  </div>
</div>
`;
